### PR TITLE
[fix]: [FFM-9542] Adds new fleature admin role

### PR DESCRIPTION
**Pull request by Andrew-Hayes**
## Describe your changes

Adds a new managed role "_feature_flag_admin_role" that includes all of the feature flag permissions.

## Checklist
- [*] I've documented the changes in the PR description.
- [*] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53716)

### CODE REVIEW

**Review by danmack-harness — APPROVED** [review]

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53716)* status: Some of the checks are too old. 2 file(s) to review. 1 change requests. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 599 at r1](https://reviewable.io/reviews/harness/harness-core/53716#-NgIvxNe0iS8k78opeML:-NgIvxNe0iS8k78opeMM:b-hq4nki) ([raw file](https://github.com/harness/harness-core/blob/58ed5cc032e52f521f5e96a7f3ede020586a0bf7/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L599)):*
> ```YAML
>   - identifier: ff_featureflag_staleReadyForCleanup
>     name: Mark a stale Feature Flag as ready for cleanup
>     status: ACTIVE
> ```

I hope you have gone through this doc
https://harness.atlassian.net/wiki/spaces/PLATFORM/pages/21060812887/Permissions+Management
___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml` line 808 at r1](https://reviewable.io/reviews/harness/harness-core/53716#-NgIv_hh9_043zgY4L3b:-NgIv_hh9_043zgY4L3c:bfxbdtv) ([raw file](https://github.com/harness/harness-core/blob/58ed5cc032e52f521f5e96a7f3ede020586a0bf7/access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml#L808)):*
> ```YAML
>         - organization
>         - project
>     - identifier: _feature_flag_admin_role
> ```

We already have _feature_flag_manage_role role. Is that needed to be different that this one?

File: access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml [hunk 1]
```diff
@@ -594,6 +594,20 @@ permissions:
       - account
       - organization
       - project
+  - identifier: ff_featureflag_staleReadyForCleanup
```
Comment (davejohnston) [inline access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:597]: should this be all snake case?

**Comment by Andrew-Hayes** [comment]
@Jimit09Gandhi yeah it was decided with product to leave the  `_feature_flag_manage_role` as it is and add a new role `_feature_flag_admin_role`. With the new admin role being more comprehensive in it's list of permissions.

**Comment by Andrew-Hayes** [comment]
We are splitting the new stale flag permissions into a separate ticket.

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53716)* status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml` line 821 at r2](https://reviewable.io/reviews/harness/harness-core/53716#-NgOfnkD9mCp_iuKh_JH:-NgOfnkD9mCp_iuKh_JI:b3juh8s) ([raw file](https://github.com/harness/harness-core/blob/fe9238820ab1819739bd0094f3d8cca756fa3136/access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml#L821)):*
> ```YAML
>         - ff_target_view
>         - ff_environment_apiKeyView
>         - ff_environment_apiKeyCreate
> ```

If this FF Admin role, shouldn't it have ff_environment_apiKeyDelete ? I see few other FF permissions missing. Please check.

**Review by Jimit09Gandhi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53716)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 597 at r1](https://reviewable.io/reviews/harness/harness-core/53716#gh-1350144744:-NgOiCbY0fLqm089L3ib:b-7crcxs) ([raw file](https://github.com/harness/harness-core/blob/58ed5cc032e52f521f5e96a7f3ede020586a0bf7/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L597)):*
<details><summary><i>Previously, davejohnston (Dave Johnston) wrote…</i></summary><blockquote>

should this be all snake case?
</blockquote></details>

 yes @davejohnston  for permission identifier

**Comment by Andrew-Hayes** [comment]
Yea it was a conscious decision to leave out the delete permissions from the role. They were deemed too dangerous even for an "admin" role

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53716)* status: 1. unresolved discussions.

**Comment by Andrew-Hayes** [comment]
trigger uts

**Comment by Andrew-Hayes** [comment]
trigger ut0

**Comment by Andrew-Hayes** [comment]
trigger ut5

**Comment by Andrew-Hayes** [comment]
trigger ut7

**Pull request merged**